### ISSUE TITLE
Rescheduling API?

### ISSUE THREAD

**Issue by jwaldrip** [coreos/fleet#1335]
Are there any plans to allow for a manual rescheduling of units? i.e. `fleetctl mv {service-id} {machine-id}`

**Comment by mischief** [comment]
@jwaldrip no, not afaik. what is the use case? why not use constraints in the unit file?

**Comment by jwaldrip** [comment]
The option here would be to have the flexibility of building my own resource based scheduler on top of what fleet has already provided. In addition, I could also preemptively move units before taking a machine down, rather than having fleet only attempt it as a recovery method. I love how low level fleet is, I just wish the API gave me more control to implement my own control plane.

**Comment by jonboulle** [comment]
On Mon, Aug 24, 2015 at 10:48 AM, Jason Waldrip <notifications@github.com>
wrote:

> fleetctl mv {service-id} {machine-id}
>
>
Well this could be implemented reasonably trivially with the equivalent of

fleetctl cat {service-id} | awk '{/MachineID=/ sub(/MachineID=.*/,
"MachineID=asdf")' > {service-id}
fleetctl destroy {service-id} && fleetctl start {service-id}

Are you looking for a way to do it through the API?

**Comment by jwaldrip** [comment]
Of course. I know it can be done with some custom scripting. By being part of the API, the move command could ensure the service has started on the new before stopping the old.

**Comment by wuqixuan** [comment]
Actually, if implement it, the code need first stop it, then start it, the logic is almost same as the script given by  jonboulle. The same unit cannot run in both machines. 
So I think the effects of the scripts and mv cmd are almost the same.

**Comment by wuqixuan** [comment]
And I think mv command is similar with overwrite command #1295, but overwrite command is more general and the functionality is more than "mv" command. So if we accept overwrite, mv duplicates a little.
